agents/ovh_cloud_rest/fence_ovh_cloud_rest.py

```diff
@@ -0,0 +1,119 @@
+            "shortdesc" : "OVH Oauth application key",
+            "order" : 1}
+    all_opt["application_secret"] = {
+            "getopt" : "s:",
+            "longopt" : "application-secret",
+            "help" : "-s, --application-secret=[application-secret]         OVH Oauth application secret",
+            "required" : "1",
+            "shortdesc" : "OVH Oauth application secret",
+            "order" : 2}
+    all_opt["token"] = {
+            "getopt" : "t:",
+            "longopt" : "token",
+            "help" : "-t, --token=[token]                                   OVH Oauth token ",
+            "required" : "1",
+            "shortdesc" : "OVH Oauth token ",
+            "order" : 3}
+    all_opt["region"] = {
+            "getopt" : "r:",
+            "longopt" : "region",
+            "help" : "-r, --region=[region]                                 OVH region (example: ovh-ca)",
+            "required" : "1",
+            "shortdesc" : "OVH region",
+            "order" : 4}
+    all_opt["service_name"] = {
+            "getopt" : "n:",
+            "longopt" : "service-name",
+            "help" : "-n, --service-name=[service-name]                     OVH service name",
+            "required" : "1",
+            "shortdesc" : "OVH service name",
+            "order" : 5}
+    all_opt["instance_id"] = {
+            "getopt" : "i:",
+            "longopt" : "instance-id",
+            "help" : "-i, --instance-id=[instance-id]                       OVH instance id",
+            "required" : "1",
+            "shortdesc" : "OVH service name",
+            "order" : 6}
+
+def ovh_login(options):
+    session = ovh.Client(
+        endpoint=options["--region"],
+        application_key=options["--application-key"],
+        application_secret=options["--application-secret"],
+        consumer_key=options["--token"],
+    )
+    options["session"] = session;
+    return session;
+
+def status(options, expected_status):
+    session = options["session"]
+    status_url = "/cloud/project/%s/instance/%s" % (options['--service-name'], options['--instance-id'])
+    response = session.get(status_url)
+    return True if response["status"] == expected_status else False
+
+def main():
+    atexit.register(atexit_handler)
+
+    device_opt = ["application_key", "application_secret", "token", "region", "service_name", "instance_id", "no_password"]
+
+    define_new_opts()
+    options = check_input(device_opt, process_input(device_opt), other_conditions=True)
+
+    docs = {}
+    docs["shortdesc"] = "Fence agent for OVH Cloud (REST API)"
+    docs["longdesc"] = "Fence agent for OVH Cloud (REST API) with authentication via Oauth"
+    docs["vendorurl"] = "https://api.ovh.com/"
+    show_docs(options, docs)
+
+    run_delay(options)
+
+    session = ovh_login(options);
+
+    rescue_url = "/cloud/project/%s/instance/%s/rescueMode" % (options['--service-name'], options['--instance-id'])
+
+    if options["--action"] == "off":
```

Comment:
You should put the on/off code into e.g. set_power_status() and make a get_power_status() where it returns on/off/unknown. This way it can e.g. send reboot as send power-off, wait for power off, send power-on, wait for power-on.

Similar agent with a good example on how to do this:
https://github.com/ClusterLabs/fence-agents/blob/master/agents/vmware_rest/fence_vmware_rest.py#L178

Reply:
Ah, good idea. I misread part of the `FenceAgentAPI` doc and thought it meant reboot was a bad idea in general (but I see it suggests that only that is a bad idea for fabric fencing).

I'll make this change.

Reply:
@oalbrigt One question -- does it matter if my `get_power_status` fails? As in, will the code in `fencing.py` retry until it gets a valid status? Or does my code need to sleep and poll until it gets a valid power status (as I've done in the code here)? It's not 100% obvious to me as I don't know if the vmware example one sometimes returns states between off and on as the OVH one does (ie if going to RESCUE, it returns RESCUING until it is ready and then RESCUE).